content/master/concepts/composition-functions.md

```diff
@@ -1,1026 +1,637 @@
-
-{{< hint "note" >}}
-The initial desired state doesn't include any copies of observed resources
-produced by the function pipeline. When using multiple functions each function
-passes their desired resources output as input to the next pipeline function.
-{{< /hint >}}
+Some people design composition functions for you to use them with any kind of
+composite resource. Function Patch and Transform and Function Auto Ready work
+with any kind of composite resource.
 
-When adding a new desired resource to the `desired.resources` array you don't
-need to:
+Another common pattern is to write a composition function specific to one kind
+of composite resource. The function contains all the logic needed to tell
+Crossplane what resources to create when you create a composite resource. When
+you write a composition function like this, your Composition can be small. It
+just tells Crossplane what function to run when you create, update, or delete a
+composite resource.
 
-* Update the XR's resource references.
-* Add any composition annotations like `crossplane.io/composite-resource-name`.
-* Set the XR as a controller/owner reference of the desired resource.
+This Composition tells Crossplane to call {{<hover label="dedicated"
+line="13">}}function-xr-xbucket{{</hover>}} whenever you create, update, or
+delete an {{<hover label="dedicated" line="8">}}XBucket{{</hover>}} composite
+resource. `function-xr-xbucket` is hard coded to handle `XBucket` composite
+resources.
 
-Crossplane will take care of all of these for you. It won't do anything else,
-including setting a sensible `metadata.name` for the new composed resource -
-this is up to your Function.
+```yaml {label="dedicated"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: example-bucket-function
+spec:
+  compositeTypeRef:
+    apiVersion: example.crossplane.io/v1
+    kind: XBucket
+  mode: Pipeline
+  pipeline:
+  - step: handle-xbucket-xr
+    functionRef:
+      name: function-xr-xbucket
+```
 
-Finally, the `results` array allows your Function to surface events and debug
-logs on the XR. Results support the following severities:
+To write a composition function, you:
 
-* `Normal` emits a debug log and a `Normal` event associated with the XR.
-* `Warning` emits a debug log and a `Warning` event associated with the XR.
-* `Fatal` stops the Composition process before applying any changes.
+1. Create the function from a template.
+1. Edit the template to add your function's logic.
+1. Test your function.
+1. Build your function, and push it to a package registry.
 
-When Crossplane encounters a `Fatal` result it will finish running the
-Composition Function pipeline. Crossplane will then return an error without
-applying any changes to the API server. Crossplane surfaces this error as a
-`Warning` event, a debug log, and by setting the `Synced` status condition of
-the XR to "False".
+You use the [Crossplane CLI](https://github.com/crossplane/docs/pull/584) to
```

Comment:
This link is a placeholder. Needs updating when https://github.com/crossplane/docs/pull/584 is merged.